mozilla-mobile/fenix#5592 - enable private browsing for custom tabs

mohamediburan opened the pull request:
This PR enables private mode for custom tabs (see issue mozilla-mobile/fenix#5592) where it will not track history. This PR already passed all the tests and I checked tabIntentProcessor.kt file for sample tests. So, I can update test for this. but, I didn't find any tests for verifying whether it a private or normal tab. If I need to write test cases for this, please let me know.

---
<!-- Text above this line will be added to the commit once "bors" merges this PR -->

### Pull Request checklist
<!-- Before submitting the PR, please address each item -->
- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

NotWoods commented:
Congrats on your first PR! It would be great if you can add a test before we merge this in.

Our `Session` object represents a tab and has a `Session#private` property to indicate if its private or not. The existing test in [`CustomTabIntentProcessorTest`](https://github.com/mozilla-mobile/android-components/blob/096e84b06b43a3cc4adf7750bddfafeaae0abf71/components/feature/customtabs/src/test/java/mozilla/components/feature/customtabs/CustomTabIntentProcessorTest.kt#L68) retrieves the newly created session and asserts on a couple properties.

I think you should update the existing test (`processCustomTabIntentWithDefaultHandlers`) to check that `customTabSession.private` is false, then add a new test that creates a private session and checks that `customTabSession.private` is true.

mohamediburan commented:
@NotWoods Thanks for reviewing. I have added the requested tests and updated an entry in the changelog.md file. Let me know if there are any changes required.

NotWoods commented:
bors r+
